pypeit/core/procimg.py

```diff
@@ -979,13 +951,13 @@ def variance_frame(datasec_img, sciframe, gain, ronoise, darkcurr=None,
     # ToDO JFH: I would just add the darkcurrent here into the effective read noise image
     # The effective read noise (variance image)
     if rnoise is None:
-        rnoise = rn_frame(datasec_img, gain, ronoise)
+        rnoise = rn2_frame(datasec_img, gain, ronoise)
 
     # No sky frame provided
     if skyframe is None:
         _darkcurr = 0 if darkcurr is None else darkcurr
         if exptime is not None:
-            _darkcurr *= exptime/3600.
+            _darkcurr *= exptime #/3600.
         var = np.abs(sciframe - np.sqrt(2.0)*np.sqrt(rnoise)) + rnoise + _darkcurr
         var = var + adderr**2*(np.abs(sciframe))**2
         return var
```

Comment:
As for this stuff below, we are computing the noise on line  in local_skysub_extract. The point is that we need to be able to construct a noise model on line 808 of core.skysub.local_skysub_extract (look for if model_noise). Ideally we would have all such code in one place. I think I tried to use this routine and then quickly got confused by the somewhat convoluted logic here,  and so opted to do just do noise computation in core.skysub.local_skysub_extract

Reply:
This part of the code now uses `pypeit.core.procimg.variance_model`.